Add obsidian-embedded-code-title plugin

tadashi-aikawa opened the pull request:
<!--- Delete this section if submitting a theme -->
# [x] I am submitting a new Community Plugin

## Repo URL

https://github.com/tadashi-aikawa/obsidian-embedded-code-title

## Release Checklist

<!--- Confirm that you have done the following before submitting your plugin -->
- [x] I have tested this on Windows, macOS, and Linux _(if applicable)_
- [x] Github release contains all required files
  - [x] `main.js`
  - [x] `manifest.json`
  - [x] `styles.css` _(optional)_
- [x] Github release name matches the exact version number specified in my manifest.json (_**Note:** Use the exact version number, don't include a prefix `v`_)
- [x] The `id` in my `manifest.json` matches the `id` in the `community-plugins.json` file.
- [x] README clearly describes the plugins purpose and provides clear usage instructions.
- [x] I have added a license in the LICENSE file

lishid commented:
https://github.com/tadashi-aikawa/obsidian-embedded-code-title/blob/2955abb498e6196f9a90c62854437b4f0e1b895f/main.ts#L16
Please rename the sample names like "MyPlugin" and "SettingTab" to real names :)

https://github.com/tadashi-aikawa/obsidian-embedded-code-title/blob/2955abb498e6196f9a90c62854437b4f0e1b895f/main.ts#L20
Please don't hook on things globally in the app like this. This may have unintended side-effects that affects non renderer panes, other plugins, etc. This also introduces a potential performance issue since you're observing pretty much the entire app's DOM.

Instead, you should use a markdown post-processor: https://github.com/obsidianmd/obsidian-api/blob/b0aa06eab03d8c39c0733949da5aa1edc2084b01/obsidian.d.ts#L2160.

https://github.com/tadashi-aikawa/obsidian-embedded-code-title/blob/2955abb498e6196f9a90c62854437b4f0e1b895f/main.ts#L87
You should make sure to cleanup any persistent events when your plugin unloads by adding an `onunload` method.

tadashi-aikawa commented:
@lishid 
Thank you for your kind review!

> Please rename the sample names like "MyPlugin" and "SettingTab" to real names :)

Fixed in https://github.com/tadashi-aikawa/obsidian-embedded-code-title/commit/aeadf870db0015b1b2adb3580046adea8b9220fc

> Please don't hook on things globally in the app like this

Fixed in https://github.com/tadashi-aikawa/obsidian-embedded-code-title/commit/e03b7e5191dd7d7c51f93e1e83dfe8c7f4f69c4d

P.S.
Actually, I had challenged to use `MarkdownPostProcessor` before PR. However, since I had misunderstood that `HTMLElement` used in `MarkdownPostProcessor` is the same with `HTMLElement` defined in `lib.dom.d.ts` , it was not working collectively. But, I have retried and worked fine. Thanks! 😄 

> You should make sure to cleanup any persistent events when your plugin unloads by adding an onunload method.

Now, I think it is not needed to clean up events. Please correct me if I'm wrong :)

lishid commented:
Looks great!

https://github.com/tadashi-aikawa/obsidian-embedded-code-title/blob/71b98fc5aeaed42faf24b9b513fb00e1a58f663f/main.ts#L67
I don't think wrapping this is necessary.

It doesn't affect the plugin much so I'll have it merged, but please give it a try without.
